multiapps-controller-web/src/main/java/org/cloudfoundry/multiapps/controller/web/api/impl/FilesApiServiceImpl.java

```diff
@@ -116,13 +120,10 @@ public ResponseEntity<FileMetadata> uploadFile(MultipartHttpServletRequest reque
         var multipartFile = getFileFromRequest(request);
         try (InputStream in = new BufferedInputStream(multipartFile.getInputStream(), INPUT_STREAM_BUFFER_SIZE)) {
             var startTime = LocalDateTime.now();
-            FileEntry fileEntry = fileService.addFile(ImmutableFileEntry.builder()
-                                                                        .space(spaceGuid)
-                                                                        .namespace(namespace)
-                                                                        .name(multipartFile.getOriginalFilename())
-                                                                        .size(BigInteger.valueOf(multipartFile.getSize()))
-                                                                        .build(),
-                                                      in);
+            FileEntry fileEntry = fileStorageThreadPool.submit(new PriorityCallable<>(PriorityFuture.Priority.LOWEST,
```

Comment:
'new PriorityCallable<>' could also be a separate method for improved readability

Reply:
done